service/discovery/discovery_test.go

```diff
@@ -172,7 +202,193 @@ func TestQuery(t *testing.T) {
 		// If a bad resource was added it will be removed. Otherwise no-op
 		delete(s.resources, "MockResourceId")
 	}
+}
+
+func TestStart(t *testing.T) {
+
+	type envVariable struct {
+		hasEnvVariable   bool
+		envVariableKey   string
+		envVariableValue string
+	}
+
+	type fields struct {
+		hasRPCConnection bool
+		envVariables     []envVariable
+		azLogout         bool
+	}
+
+	tests := []struct {
+		name           string
+		fields         fields
+		wantResp       *discovery.StartDiscoveryResponse
+		wantErr        bool
+		wantErrMessage string
+	}{
+		{
+			name: "Azure authorizer from file",
+			fields: fields{
+				hasRPCConnection: true,
+				envVariables: []envVariable{
+					{
+						hasEnvVariable:   true,
+						envVariableKey:   "AZURE_AUTH_LOCATION",
+						envVariableValue: "service/discovery/testfile/credentials_test_file",
+					},
+					{
+						hasEnvVariable:   true,
+						envVariableKey:   "HOME",
+						envVariableValue: "",
+					},
+				},
+			},
+			wantResp:       nil,
+			wantErr:        true,
+			wantErrMessage: "could not authenticate to Kubernetes",
+		},
+		{
+			name: "No Azure authorizer from file nor CLI",
+			fields: fields{
+				hasRPCConnection: true,
+				envVariables: []envVariable{
+					{
+						hasEnvVariable:   true,
+						envVariableKey:   "AZURE_AUTH_LOCATION",
+						envVariableValue: "",
+					},
+					{
+						hasEnvVariable:   true,
+						envVariableKey:   "HOME",
+						envVariableValue: "",
+					},
+				},
+			},
+			wantResp:       nil,
+			wantErr:        true,
+			wantErrMessage: "could not authenticate to Azure",
+		},
+		{
+			name: "No K8s authorizer",
+			fields: fields{
+				hasRPCConnection: true,
+				envVariables: []envVariable{
+					{
+						hasEnvVariable:   true,
+						envVariableKey:   "AZURE_AUTH_LOCATION",
+						envVariableValue: "service/discovery/testfile/credentials_test_file",
+					},
+					{
+						hasEnvVariable:   true,
+						envVariableKey:   "HOME",
+						envVariableValue: "",
+					},
+				},
+			},
+			wantResp:       nil,
+			wantErr:        true,
+			wantErrMessage: "could not authenticate to Kubernetes",
+		},
+		{
+			name: "No RPC connection",
+			fields: fields{
+				hasRPCConnection: false,
+			},
+			wantResp:       nil,
+			wantErr:        true,
+			wantErrMessage: "could not initialize stream to Assessment",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := NewService()
+			if tt.fields.hasRPCConnection {
+				assert.NoError(t, s.mockAssessmentStream())
+			}
+
+			for _, env := range tt.fields.envVariables {
+				if env.hasEnvVariable {
+					t.Setenv(env.envVariableKey, env.envVariableValue)
+				}
+			}
+
+			// We need to logout from Azure to check if no authorizer is present
+			if tt.fields.azLogout {
+				cmd := exec.Command("az", "logout")
```

Comment:
Hmm I am not sure if we really want to do this. I am not a big fan of unit tests that change the behaviour of the system we are running on.

Reply:
Ah, sorry, that was a first try. I've already do the test without the az logout and forgot to delete it.